src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs

```diff
@@ -107,13 +107,14 @@ public partial class FixAllContext
         }
 
         /// <summary>
-        /// Creates a new <see cref="FixAllContext"/> with an associated <paramref name="triggerSpan"/>.
+        /// Creates a new <see cref="FixAllContext"/> with an associated <paramref name="diagnosticSpan"/>.
         /// Use this overload when applying fix all to a diagnostic with a source location and
         /// using <see cref="FixAllScope.ContainingMember"/> or <see cref="FixAllScope.ContainingType"/>
-        /// for the <paramref name="scope"/>.  When using other fix all scopes, <paramref name="triggerSpan"/>
-        /// is not required and other constructor which does not take a triggerSpan can be used instead.
+        /// for the <paramref name="scope"/>.  When using other fix all scopes, <paramref name="diagnosticSpan"/>
+        /// is not required and other constructor which does not take a diagnostic span can be used instead.
         /// </summary>
         /// <param name="document">Document within which fix all occurrences was triggered.</param>
+        /// <param name="diagnosticSpan">Span for the diagnostic for which fix all occurrences was triggered.</param>
```

Comment:
Dumb question.  Should we just include the original diagnostic itself?  That seems 'reasonable' to me in that you started with a diagnostic, and you're asking to "fix all" of the diagnostics like it in some scope.

I basically worry that we might need some of the other data on it in the future (like teh properties/etc.) so we might just include it now for completeness.

Reply:
I had started with attempting to use diagnostic as the input, but the CodeFixContext exposes a single span and multiple possible diagnostics of the same ID for the span: https://github.com/dotnet/roslyn/blob/53468b541401e6d1fe358cc4997fcabc37ea987c/src/Workspaces/Core/Portable/CodeFixes/CodeFixContext.cs#L32-L41

Additionally, the `RegisterCodeFix` method has overloads to register a single code action fixing multiple of these diagnostics: https://github.com/dotnet/roslyn/blob/53468b541401e6d1fe358cc4997fcabc37ea987c/src/Workspaces/Core/Portable/CodeFixes/CodeFixContext.cs#L146-L166

So it seemed more logical to stick to the diagnostic span as the input (which is actually `CodeFixContext.Span` to be honest).